Source/Scene/ModelExperimental/FeatureIdPipelineStage.js

```diff
@@ -318,6 +318,13 @@ function processImplicitRange(
   );
 }
 
+const shiftsByChannelCount = [
+  "1.0",
+  "vec2(1.0, 256.0)",
+  "vec3(1.0, 256.0, 65536.0)",
+  "vec4(1.0, 256.0, 65536.0, 16777216.0)",
+];
+
 function processTexture(
   renderResources,
   featureIdTexture,
```

Comment:
In the fold below:

I realized one flaw in `processTexture()`:

right now I'm labeling the texture uniforms as `u_featureIdTexture_{featureIdSetIndex}` -- The problem is this is wasteful if two different feature ID sets share the same texture. Now you've used up 2 texture slots when only 1 is strictly needed.

I think it's better to use the glTF texture ID instead. Though I'll need to update `GltfLoader` and `ModelComponents.TextureReader` to store the index, right now it only stores the Texture object. I might do this in a future PR.

Reply:
opened https://github.com/CesiumGS/cesium/issues/10090 for this